### PR TITLE
BUG: Add messaging when removing NeighborLists from the DataStructure.

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

File: src/Plugins/SimplnxCore/src/SimplnxCore/Filters/RemoveMinimumSizeFeaturesFilter.cpp [hunk 1]
```diff
@@ -350,19 +350,19 @@ IFilter::PreflightResult RemoveMinimumSizeFeaturesFilter::preflightImpl(const Da
 
   if(applyToSinglePhase)
   {
-    const PhasesArrayType* featurePhasesPtr = data.getDataAs<PhasesArrayType>(featurePhasesPath);
-    const PhasesArrayType::store_type* featurePhases = nullptr;
+    const PhasesArrayType* featurePhasesPtr = dataStructure.getDataAs<PhasesArrayType>(featurePhasesPath);
+    // const PhasesArrayType::store_type* featurePhases = nullptr;
```
Comment (nyoungbq) [inline src/Plugins/SimplnxCore/src/SimplnxCore/Filters/RemoveMinimumSizeFeaturesFilter.cpp]:
```suggestion
```

File: src/Plugins/SimplnxCore/src/SimplnxCore/Filters/RemoveMinimumSizeFeaturesFilter.cpp [hunk 2]
```diff
@@ -350,19 +350,19 @@ IFilter::PreflightResult RemoveMinimumSizeFeaturesFilter::preflightImpl(const Da
...
     if(featurePhasesPtr != nullptr)
     {
       dataArrayPaths.push_back(featurePhasesPath);
-      featurePhases = featurePhasesPtr->getDataStore();
+      // featurePhases = featurePhasesPtr->getDataStore();
```
Comment (nyoungbq) [inline src/Plugins/SimplnxCore/src/SimplnxCore/Filters/RemoveMinimumSizeFeaturesFilter.cpp]:
```suggestion
```

File: src/simplnx/Utilities/DataGroupUtilities.cpp [hunk 3]
```diff
@@ -97,7 +98,7 @@ bool RemoveInactiveObjects(DataStructure& dataStructure, const DataPath& feature
           std::vector<DataPath> neighborListDataPaths = result.value();
           for(const auto& neighborListDataPath : neighborListDataPaths)
           {
-            dataStructure.removeData(neighborListDataPath);
+            messageHandler(nx::core::IFilter::Message{nx::core::IFilter::Message::Type::Info, fmt::format("NeighborList '{}' will be from the DataStructure.", neighborListDataPath.toString())});
```
Comment (nyoungbq) [inline src/simplnx/Utilities/DataGroupUtilities.cpp]: - This message doesn't make sense, there is no action verb.

- Also, the changelog shows there used to be a removal here that was replaced by the method, and I don't see any additions to this file that would actually remove the data. If this was intentional, I would rename the method to make clear it won't actually remove anything or add a removal function.

Comment (imikejackson) [reply src/simplnx/Utilities/DataGroupUtilities.cpp]: The removal will happen via a deferred DeleteDataAction. I am sending an information message to the user so it prints in the output log. They should have already received a warning about the NeighborLists being removed.


I will fix the sentence structure.

Comment (imikejackson) [reply src/simplnx/Utilities/DataGroupUtilities.cpp]: messageHandler(nx::core::IFilter::Message{nx::core::IFilter::Message::Type::Info, fmt::format("NeighborList '{}' will be removed from the DataStructure.", neighborListDataPath.toString())});
```suggestion
            messageHandler(nx::core::IFilter::Message{nx::core::IFilter::Message::Type::Info, fmt::format("NeighborList '{}' will be removed from the DataStructure.", neighborListDataPath.toString())});
```

**Review by nyoungbq — APPROVED** [review]

**Pull request merged**